packages/web-components/fast-components/src/radio-group/radio-group.ts

```diff
@@ -0,0 +1,254 @@
+        );
+        const inToolbar: boolean = toolbar !== undefined && toolbar !== null;
+        switch (e.keyCode) {
+            case keyCodeEnter:
+                index = this.selectedRadio ? group.indexOf(this.selectedRadio) + 1 : 1;
+                if (
+                    this.selectedRadio &&
+                    !this.selectedRadio.readOnly &&
+                    !this.selectedRadio.checked
+                ) {
+                    this.selectedRadio.checked = true;
+                    this.selectedRadio.setAttribute("tabindex", "0");
+                    this.selectedRadio.focus();
+                }
+                break;
+            case keyCodeArrowRight:
+            case keyCodeArrowDown:
+                index = this.selectedRadio ? group.indexOf(this.selectedRadio) + 1 : 1;
+                if (
+                    index === group.length &&
+                    inToolbar &&
+                    e.keyCode === keyCodeArrowRight
+                ) {
+                    this.moveRightOffGroup();
+                    return;
+                } else if (index === group.length) {
+                    index = 0;
+                }
+                // matching native radio/radiogroup which does not select an item if there is only 1 in the group
+                while (index < group.length && group.length > 1) {
+                    if (!group[index].disabled) {
+                        this.moveToRadioByIndex(group, index, inToolbar);
+                        break;
+                    } else if (index === group.indexOf(this.selectedRadio!)) {
+                        break;
+                    } else if (index + 1 >= group.length) {
+                        if (inToolbar) {
+                            break;
+                        } else {
+                            index = 0;
+                        }
+                    } else {
+                        index += 1;
+                    }
+                }
+                break;
+            case keyCodeArrowLeft:
+            case keyCodeArrowUp:
+                index = this.selectedRadio ? group.indexOf(this.selectedRadio) - 1 : 0;
+
+                if (index < 0 && inToolbar && e.keyCode === keyCodeArrowLeft) {
+                    this.moveLeftOffGroup();
+                    return;
+                }
+
+                index = index < 0 ? group.length - 1 : index;
+
+                while (index >= 0 && group.length > 1) {
+                    if (!group[index].disabled) {
+                        this.moveToRadioByIndex(group, index, inToolbar);
+                        break;
+                    } else if (index === group.indexOf(this.selectedRadio!)) {
+                        break;
+                    } else if (index - 1 < 0) {
+                        index = group.length - 1;
+                    } else {
+                        index -= 1;
+                    }
+                }
+                break;
+        }
```

Comment:
I think you need to call preventDefault() on mouse/key events you handle.

Reply:
I didn't notice this in any other web components and frankly it messes things up if I do this in radio-group. I have a question in with Rob to see if I'm missing something here to validate. Will re-work things if necessary

Reply:
confirmed with Rob, just calling preventDefault on the click, the rest of the handlers need to stay as is.